### PR TITLE
Issue 855 clarify mixin docks

### PR DESCRIPTION
**Pull request by baterson**
Hello! This is a clarification for the problem from https://github.com/pinojs/pino/issues/855

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -116,6 +116,31 @@ logger.info('world')
 // {"level":30,"time":1573664685469,"pid":78742,"hostname":"x","line":2,"msg":"world"}
 ```
 
+Be aware of providing a shared object for mixins. 
+In the case bellow, passing `mergingObject` argument to the first `info` call will mutate global `mixin` object.
```
Comment (jsumners) [inline docs/api.md:120]:
```suggestion
In the following example, passing `mergingObject` argument to the first `info` call will mutate the global `mixin` object:
```

File: docs/api.md [hunk 2]
```diff
@@ -116,6 +116,31 @@ logger.info('world')
...
+    description: 'Ok'
+}, 'Message 1')
+// {"level":30,"time":1591195061437,"pid":16012,"hostname":"x","appName":"My app","description":"Ok" "msg":"Message 1"}
+pino.info('Message 2')
+// {"level":30,"time":1591195061437,"pid":16012,"hostname":"x","appName":"My app","description":"Ok","msg":"Message 2"}
+// Note: the second log contains "description":"Ok" text, even if it was not provided.
+```
+
+If you want to just add additional info to each log message, you should look at the child loggers.
+* See [child loggers ⇗](/docs/child-loggers.md).
```
Comment (jsumners) [inline docs/api.md:142]:
```suggestion
If the `mixin` feature is being used merely to add static metadata to each log message,
then a [child logger ⇗](/docs/child-loggers.md) should be used instead.
```

**Comment by baterson** [comment]
@jsumners Thanks for helping me out!

File: docs/api.md [hunk 3]
```diff
@@ -116,6 +116,31 @@ logger.info('world')
 // {"level":30,"time":1573664685469,"pid":78742,"hostname":"x","line":2,"msg":"world"}
 ```
 
+Be aware of providing a shared object for mixins.
```
Comment (mcollina) [inline docs/api.md:119]:
```suggestion
The result of `mixin()` is supposed to be a _new_ object. For performance reason, the object returned by `mixin()` will be mutated by pino.
```

**Pull request merged**